macaroon.js

```diff
@@ -380,61 +369,62 @@ Macaroon.prototype.addCaveat = function(caveatId, verificationId, loc) {
 //
 // Verify throws an exception if the verification fails.
 Macaroon.prototype.verify = function(rootKey, check, discharges) {
-	rootKey = makeKey(rootKey)
+	rootKey = makeKey(rootKey);
 	var used = [];
 	var i;
-	for(i in discharges){
-		used[i] = 0;
+	discharges = discharges || [];
```

Comment:
This will do what we want for undefined or null in that case allowing to pass both.